File: src/main/java/com/javaToKotlin/lec04/Lec04Main.java
package com.javaToKotlin.lec04;

public class Lec04Main {

    public static void main(String[] args) {
        JavaMoney money1 = new JavaMoney(1_000L);
        JavaMoney copyMoney = money1;
        JavaMoney money2 = new JavaMoney(2_000L);

        if(money1.compareTo(money2) > 0) {
            System.out.println("Money1이 Money2보다 금액이 크다.");
        }

        System.out.println(money1 == copyMoney);    // true
        System.out.println(money1 == money2);       // false
        System.out.println(money1.equals(money2));  // true

        System.out.println(money1.plus(money2));
    }

}

File: src/main/java/com/lannstark/lec04/Lec04Main.kt
package com.lannstark.lec04

fun main() {
    val money1 = JavaMoney(2_000L)
    val copyMoney = money1
    val money2 = JavaMoney(1_000L)

    if(money1 > money2) {
        println("Money1이 Money2보다 금액이 크다.")
    }

    println(money1 === copyMoney)   // true
    println(money1 === money2)      // false
    println(money1 == money2)      // true

    if (fun1() || fun2()) { // Lazy 연산
        println("본문")
    }

    val money3 = Money(1_000L)
    val money4 = Money(2_000L)
    println(money3 + money4)    // Money(amount=3000)
}

fun fun1(): Boolean {
    println("fun 1")
    return true
}

fun fun2(): Boolean {
    println("fun 2")
    return false
}